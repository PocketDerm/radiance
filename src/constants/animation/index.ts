export default {
  defaultTiming: '350ms',
};
