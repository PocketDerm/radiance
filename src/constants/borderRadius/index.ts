const BORDER_RADIUS = {
  small: '4px',
  medium: '8px',
  large: '32px',
} as const;

export default BORDER_RADIUS;
