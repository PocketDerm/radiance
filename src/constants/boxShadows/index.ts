/*
 * Note: Each Category is exported separately to be rendered in storybook:
 */

// @ts-ignore
import { transparentize } from '../../utils';
import COLORS from '../colors';

const boxShadowColor = `${COLORS.primary}`;
const boxShadowOverlayColor = '#505050';

export const BASE_CONFIG = {
  clickable: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.06)}`,
  clickableHover: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  message: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
  focusSecondary: `0 0 2px 2px ${transparentize(COLORS.secondary, 0.5)}`,
};

export const OLD_BASE_CONFIG = {
  boxShadow1: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
  boxShadow2: `0 5px 11px 0 ${transparentize(boxShadowColor, 0.1)}`,
  boxShadow3: `0 4px 6px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow4: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow5: `0 1px 2px 0 ${transparentize(boxShadowColor, 0.06)}`,
  boxShadowOnOverlay: `0 12px 20px 0 ${boxShadowOverlayColor}`,
};

export const DESIGN_SYSTEM_NAMING = {
  elevation2: OLD_BASE_CONFIG.boxShadow5,
  elevation3: OLD_BASE_CONFIG.boxShadow4,
  elevation6: OLD_BASE_CONFIG.boxShadow3,
  elevation11: OLD_BASE_CONFIG.boxShadow2,
  elevation20: OLD_BASE_CONFIG.boxShadow1,
};

export const ALIASED_CONFIG = {
  card: OLD_BASE_CONFIG.boxShadow4,
};

export default {
  ...BASE_CONFIG,
  ...OLD_BASE_CONFIG,
  ...DESIGN_SYSTEM_NAMING,
  ...ALIASED_CONFIG,
};
