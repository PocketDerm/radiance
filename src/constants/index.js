export { default as ANIMATION } from './animation';
export { default as BOX_SHADOWS } from './boxShadows';
export { default as BREAKPOINTS } from './breakpoints';
export { default as COLORS } from './colors';
export { default as FONTS } from './fonts';
export { default as GRID } from './grid';
export { default as LAYOUT } from './layout';
export { default as MEDIA_QUERIES } from './mediaQueries';
export { default as PROGRESS_BAR_STATUS } from './progressBarStatus';
export { default as SPACING } from './spacing';
export { default as SPACER } from './spacer';
export { default as TYPOGRAPHY_CONSTANTS } from './typography';
export { default as Z_SCALE } from './zScale';
