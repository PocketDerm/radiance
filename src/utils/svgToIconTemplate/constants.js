const UTIL_LOCATION = '../../../utils/icons';

module.exports = {
  UTIL_LOCATION,
};
