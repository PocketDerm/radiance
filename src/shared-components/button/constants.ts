import { COLORS } from '../../constants';

// TODO: potentially break out these pairings to the color constants file
export const textColorsAssociatedWithColors = {
  [COLORS.primary]: {
    tint1: COLORS.primaryTint1,
    tint2: COLORS.primaryTint2,
  },
  [COLORS.secondary]: {
    tint1: COLORS.lavender80,
    tint2: COLORS.lavender60,
  },
};
