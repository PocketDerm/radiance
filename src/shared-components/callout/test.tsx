import React from 'react';
import renderer from 'react-test-renderer';

import NeckGlyph from '../../svgs/glyphs/neck-glyph.svg';
import { COLORS } from '../../constants';

import { Callout } from './index';

describe('<Callout />', () => {
  describe('UI snapshots', () => {
    it('renders a simple callout', () => {
      const component = renderer.create(
        <Callout>
          Simple callout text. <strong>Bold text supported</strong>
        </Callout>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders callout with custom color and icon', () => {
      const component = renderer.create(
        <Callout
          icon={<NeckGlyph width={40} height={40} />}
          color={COLORS.primaryTint2}
        >
          Callout text with <strong>custom color</strong> and icon.
        </Callout>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
