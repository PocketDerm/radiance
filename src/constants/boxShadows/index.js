import { transparentize } from 'src/utils';

const boxShadowColor = '#2d2d30';
const boxShadowOverlayColor = '#505050';

const BASE_CONFIG = {
  clickable: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.06)}`,
  clickableHover: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  message: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
};

const OLD_BASE_CONFIG = {
  boxShadow1: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
  boxShadow2: `0 5px 11px 0 ${transparentize(boxShadowColor, 0.1)}`,
  boxShadow3: `0 4px 6px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow4: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow5: `0 1px 2px 0 ${transparentize(boxShadowColor, 0.06)}`,
  boxShadowOnOverlay: `0 12px 20px 0 ${boxShadowOverlayColor}`,
};

const DESIGN_SYSTEM_NAMING = {
  elevation2: OLD_BASE_CONFIG.boxShadow5,
  elevation3: OLD_BASE_CONFIG.boxShadow4,
  elevation6: OLD_BASE_CONFIG.boxShadow3,
  elevation11: OLD_BASE_CONFIG.boxShadow2,
  elevation20: OLD_BASE_CONFIG.boxShadow1,
};

const ALIASED_CONFIG = {
  card: OLD_BASE_CONFIG.boxShadow4,
};

export default {
  ...BASE_CONFIG,
  ...OLD_BASE_CONFIG,
  ...DESIGN_SYSTEM_NAMING,
  ...ALIASED_CONFIG,
};

