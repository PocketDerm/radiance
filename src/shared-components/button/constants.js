import { COLORS } from 'src/constants';

//TODO: potentially break out these pairings to the color constants file
export const textColorsAssociatedWithColors = {
    purple: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    primary: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    purple100: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    lavender100: {
    	tint1: COLORS.lavender80,
    	tint2: COLORS.lavender60,
    },
    secondary: {
    	tint1: COLORS.lavender80,
    	tint2: COLORS.lavender60,
    },
};
