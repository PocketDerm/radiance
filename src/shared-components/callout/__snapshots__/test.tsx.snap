// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Callout /> UI snapshots renders a simple callout 1`] = `
.emotion-2 {
  background-color: #EEEDF7;
  padding: 1rem;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  color: #332e54;
  font-size: 0.875rem;
}

<div
  className="emotion-2 emotion-3"
>
  <div
    className="emotion-0 emotion-1"
  >
    Simple callout text. 
    <strong>
      Bold text supported
    </strong>
  </div>
</div>
`;

exports[`<Callout /> UI snapshots renders callout with custom color and icon 1`] = `
.emotion-4 {
  background-color: #EEEDF7;
  padding: 1rem;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  color: #706D87;
  font-size: 0.875rem;
}

.emotion-2 {
  margin-left: 0.5rem;
}

.emotion-2 svg {
  fill: #706D87;
}

.emotion-2 svg path {
  fill: #706D87;
}

<div
  className="emotion-4 emotion-5"
>
  <div
    className="emotion-0 emotion-1"
  >
    Callout text with 
    <strong>
      custom color
    </strong>
     and icon.
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <svg
      css={
        Object {
          "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQVE0RSIsImZpbGUiOiJzdHlsZS50cyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuXG5pbXBvcnQgeyBBTklNQVRJT04gfSBmcm9tICcuLi8uLi9jb25zdGFudHMnO1xuXG5pbXBvcnQgeyBJY29uUHJvcHMgfSBmcm9tICcuJztcblxudHlwZSBJY29uU3R5bGVzUHJvcHMgPSBQaWNrPEljb25Qcm9wcywgJ2Rpc3BsYXlJbmxpbmUnIHwgJ2ZpbGwnIHwgJ3JvdGF0ZSc+O1xuXG5jb25zdCBpY29uU3R5bGVzID0gKHsgZGlzcGxheUlubGluZSwgZmlsbCwgcm90YXRlIH06IEljb25TdHlsZXNQcm9wcykgPT4gY3NzYFxuICBkaXNwbGF5OiAke2Rpc3BsYXlJbmxpbmUgPyAnaW5saW5lLWJsb2NrJyA6ICdibG9jayd9O1xuICB0cmFuc2Zvcm06IHJvdGF0ZSgke3JvdGF0ZX1kZWcpO1xuICBjb2xvcjogJHtmaWxsfTtcbiAgdHJhbnNpdGlvbjogY29sb3IgJHtBTklNQVRJT04uZGVmYXVsdFRpbWluZ30sXG4gICAgdHJhbnNmb3JtICR7QU5JTUFUSU9OLmRlZmF1bHRUaW1pbmd9O1xuYDtcblxuZXhwb3J0IGRlZmF1bHQge1xuICBpY29uU3R5bGVzLFxufTtcbiJdfQ== */",
          "name": "vpdwsp",
          "next": undefined,
          "styles": "display:block;transform:rotate(0deg);color:;transition:color 350ms,transform 350ms;",
          "toString": [Function],
        }
      }
      height={40}
      width={40}
    />
  </div>
</div>
`;
